### ISSUE TITLE
Alternative "episode" numbers and add to NFO file.

### ISSUE THREAD

**Issue by n8detar** [meeb/tubesync#98]
I like to use the date of when the video was posted to create the season and episode of the video with the season being the year and the episode being the month plus the day: S{yyyy}E{mm}{dd}. So if something was posted on March 14, 2021, it would be S2021 E0314. Occasionally there will be videos on the same day from some channels but, at least with Jellyfin, this isn't an issue. I was wondering if you could add an option to the Add a Source page that would let you set a variable for season and episode to be added to the NFO file that is generated. So similar to Media Format, have two boxes one for the season and one for the episode. In my case, I would pet the year variable in the season box and the month and day in the episode box. I noticed that another enhancement on the road map was an episode counter, so in that case, someone could use that variable instead for the episode box.

For additional context this is what I use for my Media Format:
Season {yyyy}/{source_full} - S{yyyy}E{mm}{dd} - {title_full} - {resolution}.{ext}

**Comment by meeb** [comment]
Thanks for the suggestion. This sounds reasonable. The reason dates are currently not used for episode information in the NFO is because YouTube videos can actually have "real" season and episode numbers which should override this making this potentially a bot of a sporadic feature with unpredictable behavior. However I can see why you would want this. I also currently don't have a Jellyfin server set up, does it properly handle the episode version as mm/dd ? so S20201E03/16 is "valid" as an episode code for Jellyfin? What would you prefer if videos actually had episode codes?

This means that some channels would have the traditional S01E01 style codes where as others would have S2021E01/01 style of course.

**Comment by n8detar** [comment]
I use mmdd as the episode number without the "/" so an episode from 3/16/2021 would be episode 316 and be named with S2021E316. This format is the format I see for most media servers like Plex, Kodi, and Jellyfin and is generally recognized by them. 
I didn't know that some youtube videos would have episodes and seasons, but I think my idea would still work as you could have the default variable in the season and episode box be the value youtube gives, but then have the option to change it to a date format, or the tubesync video counter someone else proposed. So all files for that channel would be named the same way. By having variables each user can decide what format they want. 
Alternatively, you could do a specific order. So it defaults to youtube provided and if that isn't available it automatically goes to the next format which could be user selected as the date or generic video counter.

**Comment by meeb** [comment]
Yeah, all seems sensible. I'll add it to the future roadmap. Thanks for the suggestion and details.
